src/document.rs

```diff
@@ -297,6 +297,20 @@ impl Document {
         Ok(html.to_owned())
     }
 
+    /// Extracts references iff markdown content.
+    pub fn extract_markdown_references(&self) -> String {
+        let mut trail = String::new();
+        let re = Regex::new(r"(?m:^ {0,3}\[[^\]]+\]:.+$)").unwrap();
+
+        if self.markdown && re.is_match(&self.content) {
+            for mat in re.find_iter(&self.content) {
+                trail.push_str(&self.content[mat.0..mat.1]);
+                trail.push('\n');
+            }
```

Comment:
As I already have it working I'll leave it be for now.  If and when itertools becomes a dependency, this section can be revisited.